src/types/array.rs

```diff
@@ -406,10 +406,10 @@ fn build_drop<'ctx>(
 pub fn calc_data_prefix_offset(layout: Layout) -> usize {
     get_integer_layout(32)
         .extend(get_integer_layout(32))
-        .unwrap()
+        .expect("creating a layout of two i32 will never fail")
         .0
         .align_to(layout.align())
-        .unwrap()
+        .expect("only fails if size is greater than approximately ISIZE::MAX")
```

Comment:
I suggest to replace `approximately` with something a bit more specific like:

```suggestion
        .expect("only fails if layout size rounded up to the next multiple of layout alignment is greater than ISIZE::MAX")
```

You can even rename it to:

```suggestion
        .expect("layout size rounded up to the next multiple of layout alignment should never be greater than ISIZE::MAX")
```

Reply:
Done